### PR TITLE
Install xcode for git

### PR DESCRIPTION
**Pull request by cole-h**
sigh

Otherwise, the agent won't be able to clone any repos.

### CODE REVIEW

**Comment by Hoverbear** [comment]
Could we do similar to what we did with tailscale and use a nix built ssh added to path manually?

**Comment by cole-h** [comment]
(I assume you mean git) I tried, and no -- as Linus and Eelco guessed, tailscale only worked because it's mostly static (even if macOS doesn't really have an understanding of static binaries itself, Go programs don't do much external linking). Git does not behave the same:

```
ephemeraladmin@mac-epic-turducken ~ % curl -L 'https://hydra.nixos.org/job/nixpkgs/nixpkgs-unstable-aarch64-darwin/git.aarch64-darwin/latest/download/2/out/bin/git' -O                                                                         
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current                                                                                                                                                                 
                                 Dload  Upload   Total   Spent    Left  Speed                                                                                                                                                                   
100   346  100   346    0     0    597      0 --:--:-- --:--:-- --:--:--   605                                                                                                                                                                  
100 3377k  100 3377k    0     0  1907k      0  0:00:01  0:00:01 --:--:-- 3695k                                                                                                                                                                  
ephemeraladmin@mac-epic-turducken ~ % chmod +x ./git                                                                                                                                                                                            
ephemeraladmin@mac-epic-turducken ~ % ./git                                                                                                                                                                                                     
dyld[62705]: Library not loaded: '/nix/store/s6v3p7mpcvy1kdh2q35vsnwb9sva0caf-pcre2-10.40/lib/libpcre2-8.0.dylib'                                                                                                                               
  Referenced from: '/Users/ephemeraladmin/git'                                                                                                                                                                                                  
  Reason: tried: '/nix/store/s6v3p7mpcvy1kdh2q35vsnwb9sva0caf-pcre2-10.40/lib/libpcre2-8.0.dylib' (no such file), '/usr/local/lib/libpcre2-8.0.dylib' (no such file), '/usr/lib/libpcre2-8.0.dylib' (no such file)                              
zsh: abort      ./git                                                                                                                                                                                                                           

```

**Comment by Hoverbear** [comment]
Right and we don't build `pkgsStatic.git` on hydra. :(

File: setup-without-nix.sh [hunk 1]
```diff
@@ -41,6 +41,16 @@ set -o pipefail
...
   fi
 
+  # install xcode (for git) ugh
+  # inspired by https://gist.github.com/mokagio/b974620ee8dcf5c0671f
+  # and yes, this file is required, or else the command line tools don't show up in softwareupdate -l
+  touch /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress
+  softwareupdate -i "$(softwareupdate -l | grep "\*.*Command Line" | tail -n 1 | sed 's/^[^C]* //')"
+  if ! git --help &>/dev/null; then
+    # Didn't find command line tools first time, try again?
+    softwareupdate -i "$(softwareupdate -l | grep "\*.*Command Line" | tail -n 1 | sed 's/^[^C]* //')"
```
Comment (Hoverbear) [inline setup-without-nix.sh:51]: Mmmm does this... help?

Comment (cole-h) [reply setup-without-nix.sh:51]: Yes. One time, I witnessed a machine that booted with this commit, but still did not have `git` installed. So we try again just once.

**Comment by cole-h** [comment]
We're not enthused about installing xcode, but it's "necessary" for git :')

(Graham gave me the go-ahead in DMs)

**Pull request merged**